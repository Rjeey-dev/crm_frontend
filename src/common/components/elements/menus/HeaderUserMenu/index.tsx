import React, {Component} from 'react';

import {
    ABILITY_LOGOUT_LINK,
    ABILITY_SEE
} from "client/ability";
import LogOutLink from 'client/containers/links/LogOut/index';
import Ability from "containers/common/Ability";

interface IProps {
    classes?: string
}

class HeaderUserMenu extends Component<IProps> {
    constructor(props: IProps) {
        super(props);

        this.handleOver = this.handleOver.bind(this);
        this.handleClick = this.handleClick.bind(this);
    }

    public handleOver(event: any) {
        event.preventDefault();

        this.openBlock();
    }

    public handleOut(event: any) {
        event.preventDefault();
        const wrapper = document.getElementById('dropdown-menu-user-menu');

        if (wrapper) {
            wrapper.classList.remove('active');
        }
    }

    public handleClick(event: any) {
        event.preventDefault();

        this.openBlock();
    }

    private openBlock() {
        const wrapper = document.getElementById('dropdown-menu-user-menu');

        if (wrapper) {
            wrapper.classList.add('active');
        }
    }

    public render() {
        return <div className='menu-item-wrapper' onClick={this.handleClick} onMouseOver={this.handleOver} onMouseLeave={this.handleOut}>
            <div className="dropdown-menu" id='dropdown-menu-user-menu'>
                <Ability run={ABILITY_SEE} on={ABILITY_LOGOUT_LINK}>
                    <LogOutLink/>
                </Ability>
            </div>
        </div>
    }
}

export default HeaderUserMenu;