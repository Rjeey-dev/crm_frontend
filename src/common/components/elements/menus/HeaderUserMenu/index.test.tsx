import { shallow } from 'enzyme';
import React from 'react';

import HeaderUserMenu from "./index";

jest.mock('containers/links/LogOut', () => 'log-out-link');

it("Renders <HeaderUserMenu/> with correct properties", () => {
    expect(
        shallow(<HeaderUserMenu id='header-user-menu' classes='classes'/>).html()
    ).toMatchSnapshot();
});

it("Renders <HeaderUserMenu/> with only required props", () => {
    expect(
        shallow(<HeaderUserMenu id='header-user-menu'/>).html()
    ).toMatchSnapshot();
});