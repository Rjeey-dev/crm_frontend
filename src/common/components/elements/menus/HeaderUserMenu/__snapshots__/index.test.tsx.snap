// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <HeaderUserMenu/> with correct properties 1`] = `"<div><a href=\\"#\\" id=\\"header-user-menu\\" class=\\"classes\\" aria-expanded=\\"false\\" data-toggle=\\"dropdown\\"><span class=\\"profile-text\\">Missing translationId: common.hello_with_name for language: \${ languageCode }</span><img class=\\"img-xs rounded-circle\\" src=\\"images/faces/face1.jpg\\" alt=\\"Profile image\\"/></a><div class=\\"dropdown-menu dropdown-menu-right navbar-dropdown\\"><div class=\\"d-flex border-bottom\\"><div class=\\"py-3 px-4 d-flex align-items-center justify-content-center\\"><a href=\\"#\\"><i class=\\"mdi mdi-bookmark-plus-outline mr-0 text-gray\\"></i></a></div><div class=\\"py-3 px-4 d-flex align-items-center justify-content-center border-left border-right\\"><a href=\\"#\\"><i class=\\"mdi mdi-account-outline mr-0 text-gray\\"></i></a></div><div class=\\"py-3 px-4 d-flex align-items-center justify-content-center\\"><a href=\\"#\\"><i class=\\"mdi mdi-alarm-check mr-0 text-gray\\"></i></a></div></div><a href=\\"#/profile\\" class=\\"dropdown-item mt-2\\">Missing translationId: menus.header_user_menu.manage_account for language: \${ languageCode }</a><a href=\\"#\\" class=\\"dropdown-item\\">Missing translationId: menus.header_user_menu.change_password for language: \${ languageCode }</a><a href=\\"#\\" class=\\"dropdown-item\\">Missing translationId: menus.header_user_menu.check_inbox for language: \${ languageCode }</a><log-out-link></log-out-link></div></div>"`;

exports[`Renders <HeaderUserMenu/> with only required props 1`] = `"<div><a href=\\"#\\" id=\\"header-user-menu\\" aria-expanded=\\"false\\" data-toggle=\\"dropdown\\"><span class=\\"profile-text\\">Missing translationId: common.hello_with_name for language: \${ languageCode }</span><img class=\\"img-xs rounded-circle\\" src=\\"images/faces/face1.jpg\\" alt=\\"Profile image\\"/></a><div class=\\"dropdown-menu dropdown-menu-right navbar-dropdown\\"><div class=\\"d-flex border-bottom\\"><div class=\\"py-3 px-4 d-flex align-items-center justify-content-center\\"><a href=\\"#\\"><i class=\\"mdi mdi-bookmark-plus-outline mr-0 text-gray\\"></i></a></div><div class=\\"py-3 px-4 d-flex align-items-center justify-content-center border-left border-right\\"><a href=\\"#\\"><i class=\\"mdi mdi-account-outline mr-0 text-gray\\"></i></a></div><div class=\\"py-3 px-4 d-flex align-items-center justify-content-center\\"><a href=\\"#\\"><i class=\\"mdi mdi-alarm-check mr-0 text-gray\\"></i></a></div></div><a href=\\"#/profile\\" class=\\"dropdown-item mt-2\\">Missing translationId: menus.header_user_menu.manage_account for language: \${ languageCode }</a><a href=\\"#\\" class=\\"dropdown-item\\">Missing translationId: menus.header_user_menu.change_password for language: \${ languageCode }</a><a href=\\"#\\" class=\\"dropdown-item\\">Missing translationId: menus.header_user_menu.check_inbox for language: \${ languageCode }</a><log-out-link></log-out-link></div></div>"`;
